### ISSUE TITLE
Could not register device plugin: context deadline exceeded

### ISSUE THREAD

**Issue by mfojtak** [NVIDIA/k8s-device-plugin#10]
I am getting following error when starting the plugin as docker container

```
2017/11/24 09:06:24 Loading NVML
2017/11/24 09:06:24 Starting to serve on /var/lib/kubelet/device-plugins/nvidia.sock
2017/11/24 09:06:29 Could not register device plugin: context deadline exceeded
```
My installation of nvidia-docker works fine.

What is the problem?

**Comment by RenaudWasTaken** [comment]
Hello @mfojtak !

Can you attach the kubelet logs to this too ? It looks like the Kubelet's device plugin gRPC server is not up.

Thanks!

**Comment by mfojtak** [comment]
These log records seem to be related:

```
Nov 24 13:17:41 -ubuntu16 kubelet[1154]: W1124 13:17:41.949553    1154 cni.go:265] CNI failed to retrieve network namespace path: Cannot find network namespace for the terminated container "3d8582cef73e93dcfbc7ed4d73ba9483b3ac5254adf6870c0949becd5f63df99"
Nov 24 13:17:34 -ubuntu16 kubelet[1154]: W1124 13:17:34.439552    1154 helpers.go:847] eviction manager: no observation found for eviction signal allocatableNodeFs.available
Nov 24 13:17:24 -ubuntu16 kubelet[1154]: W1124 13:17:24.409622    1154 helpers.go:847] eviction manager: no observation found for eviction signal allocatableNodeFs.available
Nov 24 13:17:22 -ubuntu16 kubelet[1154]: I1124 13:17:22.259828    1154 reconciler.go:212] operationExecutor.VerifyControllerAttachedVolume started for volume "default-token-9z9jw" (UniqueName: "kubernetes.io/secret/ce2e6f51-d119-11e7-ab5c-005056990f37-default-token-9z9jw") pod "nvidia-device-plugin-daemonset-nlbzb" (UID: "ce2e6f51-d119-11e7-ab5c-005056990f37")
Nov 24 13:17:22 -ubuntu16 kubelet[1154]: I1124 13:17:22.259749    1154 reconciler.go:212] operationExecutor.VerifyControllerAttachedVolume started for volume "device-plugin" (UniqueName: "kubernetes.io/host-path/ce2e6f51-d119-11e7-ab5c-005056990f37-device-plugin") pod "nvidia-device-plugin-daemonset-nlbzb" (UID: "ce2e6f51-d119-11e7-ab5c-005056990f37")
Nov 24 13:17:22 -ubuntu16 kubelet[1154]: E1124 13:17:22.204683    1154 helpers.go:468] PercpuUsage had 0 cpus, but the actual number is 24; ignoring extra CPUs
Nov 24 13:17:22 -ubuntu16 kubelet[1154]: weave-cni: unable to release IP address: 400 Bad Request: Delete: no addresses for 3d8582cef73e93dcfbc7ed4d73ba9483b3ac5254adf6870c0949becd5f63df99
Nov 24 13:17:22 -ubuntu16 kubelet[1154]: W1124 13:17:22.088475    1154 cni.go:265] CNI failed to retrieve network namespace path: Cannot find network namespace for the terminated container "3d8582cef73e93dcfbc7ed4d73ba9483b3ac5254adf6870c0949becd5f63df99"
Nov 24 13:17:21 -ubuntu16 kubelet[1154]: I1124 13:17:21.257160    1154 reconciler.go:290] Volume detached for volume "default-token-9z9jw" (UniqueName: "kubernetes.io/secret/8f158ac2-d115-11e7-ab5c-005056990f37-default-token-9z9jw") on node "-ubuntu16" DevicePath ""
Nov 24 13:17:21 -ubuntu16 kubelet[1154]: I1124 13:17:21.191011    1154 operation_generator.go:535] UnmountVolume.TearDown succeeded for volume "kubernetes.io/secret/8f158ac2-d115-11e7-ab5c-005056990f37-default-token-9z9jw" (OuterVolumeSpecName: "default-token-9z9jw") pod "8f158ac2-d115-11e7-ab5c-005056990f37" (UID: "8f158ac2-d115-11e7-ab5c-005056990f37"). InnerVolumeSpecName "default-token-9z9jw". PluginName "kubernetes.io/secret", VolumeGidValue ""
Nov 24 13:17:21 -ubuntu16 kubelet[1154]: I1124 13:17:21.156821    1154 reconciler.go:290] Volume detached for volume "device-plugin" (UniqueName: "kubernetes.io/host-path/8f158ac2-d115-11e7-ab5c-005056990f37-device-plugin") on node "-ubuntu16" DevicePath ""
Nov 24 13:17:21 -ubuntu16 kubelet[1154]: I1124 13:17:21.156729    1154 reconciler.go:186] operationExecutor.UnmountVolume started for volume "default-token-9z9jw" (UniqueName: "kubernetes.io/secret/8f158ac2-d115-11e7-ab5c-005056990f37-default-token-9z9jw") pod "8f158ac2-d115-11e7-ab5c-005056990f37" (UID: "8f158ac2-d115-11e7-ab5c-005056990f37")
Nov 24 13:17:21 -ubuntu16 kubelet[1154]: I1124 13:17:21.156684    1154 operation_generator.go:535] UnmountVolume.TearDown succeeded for volume "kubernetes.io/host-path/8f158ac2-d115-11e7-ab5c-005056990f37-device-plugin" (OuterVolumeSpecName: "device-plugin") pod "8f158ac2-d115-11e7-ab5c-005056990f37" (UID: "8f158ac2-d115-11e7-ab5c-005056990f37"). InnerVolumeSpecName "device-plugin". PluginName "kubernetes.io/host-path", VolumeGidValue ""
Nov 24 13:17:21 -ubuntu16 kubelet[1154]: I1124 13:17:21.156618    1154 reconciler.go:186] operationExecutor.UnmountVolume started for volume "device-plugin" (UniqueName: "kubernetes.io/host-path/8f158ac2-d115-11e7-ab5c-005056990f37-device-plugin") pod "8f158ac2-d115-11e7-ab5c-005056990f37" (UID: "8f158ac2-d115-11e7-ab5c-005056990f37")
Nov 24 13:17:21 -ubuntu16 kubelet[1154]: weave-cni: unable to release IP address: 400 Bad Request: Delete: no addresses for 3d8582cef73e93dcfbc7ed4d73ba9483b3ac5254adf6870c0949becd5f63df99
Nov 24 13:17:21 -ubuntu16 kubelet[1154]: W1124 13:17:21.059238    1154 cni.go:265] CNI failed to retrieve network namespace path: Cannot find network namespace for the terminated container "3d8582cef73e93dcfbc7ed4d73ba9483b3ac5254adf6870c0949becd5f63df99"
Nov 24 13:17:21 -ubuntu16 kubelet[1154]: W1124 13:17:21.056435    1154 pod_container_deletor.go:77] Container "3d8582cef73e93dcfbc7ed4d73ba9483b3ac5254adf6870c0949becd5f63df99" not found in pod's containers
Nov 24 13:17:20 -ubuntu16 kubelet[1154]:  with error: exit status 1
Nov 24 13:17:20 -ubuntu16 kubelet[1154]: W1124 13:17:20.017189    1154 docker_sandbox.go:343] failed to read pod IP from plugin/docker: NetworkPlugin cni failed on the status hook for pod "nvidia-device-plugin-daemonset-qhmzz_default": Unexpected command output nsenter: cannot open /proc/102125/ns/net: No such file or directory
```

Thank you

**Comment by mfojtak** [comment]
I found what the problem was.

I did not enable DevicePlugins feature gate in kubelet.

It requires to edit
```
/etc/systemd/system/kubelet.service.d/10-kubeadm.conf
```
and add
```
--feature-gates=DevicePlugins=true
```

Maybe it would be good to put a warning into the docs about is.

Thank you anyway. The gRPC not running comment pointed me to the right direction.

**Comment by RenaudWasTaken** [comment]
Thanks! Will be updating the logs :)

**Comment by flx42** [comment]
Fixed by: 8848bbb925c8f860037c62b3c77f00c738924f09

@RenaudWasTaken next time, please add `Closes: #10` to your commit message ;)
